tools/amazon_agent_settings/prepare_userdata

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
```

Comment:
This whole script is very confusing.  It feels like you basically took the ruby code that generated the userdata and ported it to bash, but the request was to just create raw script files and then just have some Ruby code to assemble / interpolate it.  In fact, you could use erb to do the interpolation to make it even easier.

Reply:
I'm sure I'm missing something because of the weird layers of indirection due to the usage of userdata...let's discuss f2f.

Reply:
For the script, let's discuss it f2f when you have time.